File: CareConnectApp/src/screens/ASLHelpScreen.js
import React, { useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Pressable,
  ScrollView,
} from 'react-native';
import {
  Play,
  Pause,
  Volume2,
  VolumeX,
  Subtitles,
  ChevronRight,
} from 'lucide-react-native';

const ASLHelpScreen = ({ onNavigate }) => {
  const [isPlaying, setIsPlaying] = useState(false);
  const [isMuted, setIsMuted] = useState(false);
  const [showCaptions, setShowCaptions] = useState(true);

  const aslVideos = [
    { id: 1, title: 'How to Take Your Medication', duration: '2:45', emoji: '💊' },
    { id: 2, title: 'Understanding Your Daily Tasks', duration: '3:20', emoji: '📋' },
    { id: 3, title: 'Using the SOS Button', duration: '1:30', emoji: '🆘' },
    { id: 4, title: 'Setting Up Reminders', duration: '2:15', emoji: '⏰' },
  ];

  return (
    <ScrollView style={styles.container}>
      {/* Featured Video */}
      <View style={styles.videoCard}>
        <View style={styles.videoDisplay}>
          <Text style={styles.aslEmoji}>🤟</Text>
          <Text style={styles.videoTitle}>ASL Interpretation</Text>

          {showCaptions && (
            <View style={styles.captionOverlay}>
              <Text style={styles.captionText}>
                Welcome to CareConnect. This video will show you how to take your
                medication safely.
              </Text>
            </View>
          )}
        </View>

        {/* Controls */}
        <View style={styles.controls}>
          <View style={styles.controlRow}>
            <Pressable
              style={styles.playButton}
              onPress={() => setIsPlaying(!isPlaying)}
            >
              {isPlaying ? (
                <Pause size={24} color="#FFFFFF" />
              ) : (
                <Play size={24} color="#FFFFFF" />
              )}
            </Pressable>

            <View style={styles.controlIcons}>
              <Pressable onPress={() => setIsMuted(!isMuted)}>
                {isMuted ? (
                  <VolumeX size={24} color="#FFFFFF" />
                ) : (
                  <Volume2 size={24} color="#FFFFFF" />
                )}
              </Pressable>

              <Pressable onPress={() => setShowCaptions(!showCaptions)}>
                <Subtitles
                  size={24}
                  color={showCaptions ? '#60A5FA' : '#FFFFFF'}
                />
              </Pressable>
            </View>
          </View>

          {/* Progress */}
          <View style={styles.progressBar}>
            <View style={styles.progressFill} />
          </View>

          <View style={styles.timeRow}>
            <Text style={styles.timeText}>0:45</Text>
            <Text style={styles.timeText}>2:45</Text>
          </View>
        </View>
      </View>

      <Text style={styles.sectionTitle}>How to Take Your Medication</Text>
      <Text style={styles.sectionSubtitle}>
        Learn the step-by-step process with ASL interpretation and captions
      </Text>

      {/* Video Library */}
      <Text style={styles.libraryTitle}>More Help Videos</Text>

      {aslVideos.map((video) => (
        <Pressable key={video.id} style={styles.videoRow}>
          <View style={styles.thumbnail}>
            <Text style={styles.thumbnailEmoji}>{video.emoji}</Text>
          </View>

          <View style={{ flex: 1 }}>
            <Text style={styles.videoRowTitle}>{video.title}</Text>
            <View style={styles.badgeRow}>
              <Text style={styles.badge}>ASL + Captions</Text>
              <Text style={styles.duration}>{video.duration}</Text>
            </View>
          </View>

          <ChevronRight size={20} color="#94A3B8" />
        </Pressable>
      ))}

      {/* Accessibility Info */}
      <View style={styles.infoCard}>
        <Text style={styles.infoText}>
          ♿ <Text style={{ fontWeight: '700' }}>Accessibility:</Text> All videos
          include American Sign Language (ASL) interpretation and closed
          captions. Captions can be toggled at any time.
        </Text>
      </View>
    </ScrollView>
  );
};

export default ASLHelpScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F8FAFC',
    padding: 16,
  },

  videoCard: {
    backgroundColor: '#0F172A',
    borderRadius: 20,
    overflow: 'hidden',
    marginBottom: 24,
  },

  videoDisplay: {
    height: 220,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#020617',
  },

  aslEmoji: {
    fontSize: 72,
    marginBottom: 8,
  },

  videoTitle: {
    color: '#FFFFFF',
    fontSize: 18,
    fontWeight: '700',
  },

  captionOverlay: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    backgroundColor: 'rgba(0,0,0,0.8)',
    padding: 12,
  },

  captionText: {
    color: '#FFFFFF',
    textAlign: 'center',
    fontSize: 14,
  },

  controls: {
    backgroundColor: '#1E293B',
    padding: 16,
  },

  controlRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 12,
  },

  playButton: {
    backgroundColor: '#2563EB',
    padding: 12,
    borderRadius: 999,
  },

  controlIcons: {
    flexDirection: 'row',
    gap: 16,
  },

  progressBar: {
    height: 6,
    backgroundColor: '#334155',
    borderRadius: 999,
    overflow: 'hidden',
  },

  progressFill: {
    width: '33%',
    height: '100%',
    backgroundColor: '#3B82F6',
  },

  timeRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 6,
  },

  timeText: {
    color: '#E5E7EB',
    fontSize: 12,
  },

  sectionTitle: {
    fontSize: 20,
    fontWeight: '700',
    color: '#0F172A',
    marginBottom: 4,
  },

  sectionSubtitle: {
    color: '#64748B',
    fontSize: 13,
    marginBottom: 20,
  },

  libraryTitle: {
    fontSize: 18,
    fontWeight: '700',
    marginBottom: 12,
    color: '#0F172A',
  },

  videoRow: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
    borderRadius: 14,
    padding: 12,
    borderWidth: 2,
    borderColor: '#E5E7EB',
    marginBottom: 10,
    gap: 12,
  },

  thumbnail: {
    width: 64,
    height: 64,
    backgroundColor: '#DBEAFE',
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
  },

  thumbnailEmoji: {
    fontSize: 28,
  },

  videoRowTitle: {
    fontWeight: '600',
    color: '#0F172A',
    marginBottom: 4,
  },

  badgeRow: {
    flexDirection: 'row',
    gap: 8,
  },

  badge: {
    backgroundColor: '#DBEAFE',
    color: '#2563EB',
    fontSize: 11,
    fontWeight: '600',
    paddingHorizontal: 8,
    paddingVertical: 2,
    borderRadius: 999,
  },

  duration: {
    fontSize: 11,
    color: '#64748B',
  },

  infoCard: {
    backgroundColor: '#EFF6FF',
    borderColor: '#BFDBFE',
    borderWidth: 2,
    borderRadius: 14,
    padding: 16,
    marginTop: 20,
    marginBottom: 32,
  },

  infoText: {
    color: '#1E40AF',
    fontSize: 13,
  },
});

File: careconnect_mobile/lib/screens/asl_help_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../widgets/bottom_navigation_bar.dart';

class ASLHelpScreen extends StatefulWidget {
  const ASLHelpScreen({super.key});

  @override
  State<ASLHelpScreen> createState() => _ASLHelpScreenState();
}

class _ASLHelpScreenState extends State<ASLHelpScreen> {
  bool isPlaying = false;
  bool isMuted = false;
  bool showCaptions = true;

  final aslVideos = const [
    {
      "title": "How to Take Your Medication",
      "duration": "2:45",
      "emoji": "💊",
    },
    {
      "title": "Understanding Your Daily Tasks",
      "duration": "3:20",
      "emoji": "📋",
    },
    {
      "title": "Using the SOS Button",
      "duration": "1:30",
      "emoji": "🆘",
    },
    {
      "title": "Setting Up Reminders",
      "duration": "2:15",
      "emoji": "⏰",
    },
  ];

  @override
  Widget build(BuildContext context) {
    final isTablet = MediaQuery.of(context).size.width > 700;
    final isLandscape =
        MediaQuery.of(context).orientation == Orientation.landscape;

    return Scaffold(
      appBar: AppBar(
        leading: IconButton(
          icon: const Icon(Icons.arrow_back),
          onPressed: () {
          if (context.canPop()) {
            context.pop();
          } else {
            context.go('/dashboard');
          }
        },
          tooltip: 'Back',
        ),
        title: const Text("ASL Help Videos"),
      ),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          _featuredPlayer(isTablet),

          const SizedBox(height: 12),

          const Text(
            "How to Take Your Medication",
            style: TextStyle(fontSize: 22, fontWeight: FontWeight.bold),
          ),
          const SizedBox(height: 4),
          const Text(
            "Learn the step-by-step process with ASL interpretation and captions",
            style: TextStyle(color: Colors.grey),
          ),

          const SizedBox(height: 28),

          const Text(
            "More Help Videos",
            style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
          ),

          const SizedBox(height: 12),

          GridView.builder(
            shrinkWrap: true,
            physics: const NeverScrollableScrollPhysics(),
            gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
              crossAxisCount: isTablet && isLandscape ? 2 : 1,
              mainAxisSpacing: 12,
              crossAxisSpacing: 12,
              childAspectRatio: 3.5,
            ),
            itemCount: aslVideos.length,
            itemBuilder: (context, index) {
              return _videoTile(aslVideos[index]);
            },
          ),

          const SizedBox(height: 24),

          _accessibilityInfo(),
        ],
      ),
      bottomNavigationBar: CareConnectBottomNavBar(
        currentRoute: GoRouter.of(context).routerDelegate.currentConfiguration.uri.path,
      ),
    );
  }

  // ===============================
  // FEATURED PLAYER
  // ===============================

  Widget _featuredPlayer(bool isTablet) {
    return Container(
      decoration: BoxDecoration(
        color: Colors.grey.shade900,
        borderRadius: BorderRadius.circular(20),
        boxShadow: const [
          BoxShadow(color: Colors.black26, blurRadius: 10),
        ],
      ),
      child: Column(
        children: [
          AspectRatio(
            aspectRatio: 16 / 9,
            child: Stack(
              alignment: Alignment.bottomCenter,
              children: [
                Container(
                  decoration: const BoxDecoration(
                    gradient: LinearGradient(
                      colors: [Colors.indigo, Colors.black87],
                    ),
                  ),
                  child: Center(
                    child: Column(
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        Text(
                          "🤟",
                          style: TextStyle(
                            fontSize: isTablet ? 90 : 64,
                          ),
                        ),
                        const SizedBox(height: 8),
                        const Text(
                          "ASL Interpretation",
                          style: TextStyle(
                              color: Colors.white,
                              fontWeight: FontWeight.w600),
                        ),
                      ],
                    ),
                  ),
                ),

                if (showCaptions)
                  Container(
                    width: double.infinity,
                    padding: const EdgeInsets.all(12),
                    color: Colors.black87,
                    child: Text(
                      "Welcome to CareConnect. This video will show you how to take your medication safely.",
                      textAlign: TextAlign.center,
                      style: TextStyle(
                        color: Colors.white,
                        fontSize: isTablet ? 18 : 14,
                      ),
                    ),
                  ),
              ],
            ),
          ),

          // CONTROLS
          Container(
            color: Colors.grey.shade800,
            padding: const EdgeInsets.all(14),
            child: Column(
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Semantics(
                      label: isPlaying ? 'Pause video' : 'Play video',
                      button: true,
                      child: IconButton(
                        iconSize: 36,
                        color: Colors.white,
                        onPressed: () {
                          setState(() => isPlaying = !isPlaying);
                        },
                        icon: Icon(
                            isPlaying ? Icons.pause : Icons.play_arrow),
                      ),
                    ),
                    Row(
                      children: [
                        Semantics(
                          label: isMuted ? 'Unmute audio' : 'Mute audio',
                          button: true,
                          child: IconButton(
                            icon: Icon(isMuted
                                ? Icons.volume_off
                                : Icons.volume_up),
                            color: Colors.white,
                            onPressed: () {
                              setState(() => isMuted = !isMuted);
                            },
                          ),
                        ),
                        Semantics(
                          label: showCaptions ? 'Hide captions' : 'Show captions',
                          button: true,
                          selected: showCaptions,
                          child: IconButton(
                            icon: Icon(Icons.subtitles,
                                color: showCaptions
                                    ? Colors.blue
                                    : Colors.white),
                            onPressed: () {
                              setState(() => showCaptions = !showCaptions);
                            },
                          ),
                        ),
                      ],
                    )
                  ],
                ),

                const SizedBox(height: 6),

                LinearProgressIndicator(
                  value: .33,
                  backgroundColor: Colors.grey.shade700,
                ),

                const SizedBox(height: 4),

                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: const [
                    Text("0:45",
                        style: TextStyle(color: Colors.white70)),
                    Text("2:45",
                        style: TextStyle(color: Colors.white70)),
                  ],
                )
              ],
            ),
          )
        ],
      ),
    );
  }

  // ===============================
  // VIDEO TILE
  // ===============================

  Widget _videoTile(Map<String, String> video) {
    final videoLabel = '${video["title"]!}, ${video["duration"]!}, ASL and captions available';
    return Semantics(
      label: videoLabel,
      button: true,
      hint: 'Tap to play video',
      child: Container(
        padding: const EdgeInsets.all(12),
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(16),
          border: Border.all(color: Colors.grey.shade300),
        ),
        child: Row(
          children: [
            Semantics(
              label: 'Video thumbnail',
              image: true,
              excludeSemantics: true,
              child: Container(
                width: 64,
                height: 64,
                decoration: BoxDecoration(
                  borderRadius: BorderRadius.circular(12),
                  gradient: LinearGradient(
                    colors: [Colors.blue.shade100, Colors.blue.shade200],
                  ),
                ),
                child: Center(
                  child: Text(video["emoji"]!,
                      style: const TextStyle(fontSize: 28)),
                ),
              ),
            ),

            const SizedBox(width: 12),

            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Text(
                    video["title"]!,
                    style: const TextStyle(fontWeight: FontWeight.w600),
                  ),
                  const SizedBox(height: 6),
                  Row(
                    children: [
                      Container(
                        padding: const EdgeInsets.symmetric(
                            horizontal: 8, vertical: 4),
                        decoration: BoxDecoration(
                          color: Colors.blue.shade100,
                          borderRadius: BorderRadius.circular(12),
                        ),
                        child: const Text(
                          "ASL + Captions",
                          style: TextStyle(
                              fontSize: 12,
                              color: Colors.blue),
                        ),
                      ),
                      const SizedBox(width: 8),
                      Text(video["duration"]!,
                          style: const TextStyle(
                              fontSize: 12, color: Colors.grey)),
                    ],
                  ),
                ],
              ),
            ),

            Semantics(
              label: 'Play video',
              image: true,
              excludeSemantics: true,
              child: const Icon(Icons.chevron_right, color: Colors.grey),
            ),
          ],
        ),
      ),
    );
  }

  // ===============================
  // ACCESSIBILITY INFO
  // ===============================

  Widget _accessibilityInfo() {
    return Container(
      padding: const EdgeInsets.all(14),
      decoration: BoxDecoration(
        color: Colors.blue.shade50,
        borderRadius: BorderRadius.circular(14),
        border: Border.all(color: Colors.blue.shade200),
      ),
      child: const Text(
        "♿ Accessibility: All videos include American Sign Language (ASL) interpretation and closed captions. You can toggle captions on/off using the subtitle button.",
        style: TextStyle(fontSize: 13),
      ),
    );
  }
}
